### DISCUSSION TITLE
Image Size

### DISCUSSION THREAD

**Discussion by st162053** [facebookresearch/detectron2#3942] [Q&A]
Hi, 
is it possible to use different image sizes for training a custom dataset?
I have a lot of different sizes, because of using different cameras for generating my dataset.
Thanks!

**Comment by zhanghang1989** [comment]
It is automatically handled by `ImageList.from_tensors()`

**Reply by andreaceruti** [reply]
@zhanghang1989 Can you tell me in which file it is handled? Do you think that it is better to resize them in order to have a square image or this will reduce the imahe information?

**Reply by zhanghang1989** [reply]
https://github.com/facebookresearch/detectron2/blob/main/detectron2/modeling/meta_arch/rcnn.py#L230

**Comment by edehino** [comment]
@zhanghang1989 I am preparing my own dataset. Currently, I have 1500x1500 image size. Is it fine to use this size? Is there any recommended image size to get a good result in training?

**Reply by zhanghang1989** [reply]
The image size may be too large, which may consume too much memory during the training. It is fine to prepare a dataset with that size, you can adjust the training size later.

**Reply by edehino** [reply]
Is there any way I can resize the annotated images? When you say training size, do you mean batch size or the numbet of images to train?
